arbor/mc_cell_group.hpp

```diff
@@ -57,10 +57,26 @@ class mc_cell_group: public cell_group {
 
     void remove_all_samplers() override;
 
+    void generate_deps_gids(const recipe& rec, std::vector<cell_gid_type>);
+
+    std::vector<cell_gid_type> get_gids() {
+        return gids_;
+    }
+
+    std::vector<int> get_dependencies() {
+        return deps_;
+    }
+
 private:
     // List of the gids of the cells in the group.
     std::vector<cell_gid_type> gids_;
 
+    // Permutation of the gids of the cells in the group.
+    std::vector<cell_gid_type> perm_gids_;
```

Comment:
We're not always consistent on this, but we should try to keep `cell_gid_type` for values that are actual gids. I'd use `cell_size_type` in preference (which is meant to be an unsigned value for counting cells/gids.)

Reply:
Okay.